distributor/distributor.go

```diff
@@ -361,6 +367,9 @@ func (d *Distributor) sendSamples(ctx context.Context, ingester *ring.IngesterDe
 	}
 	err = instrument.TimeRequestHistogram(ctx, "Distributor.sendSamples", d.sendDuration, func(ctx context.Context) error {
 		_, err := client.Push(ctx, util.ToWriteRequest(samples))
+		if grpc.Code(err) == codes.ResourceExhausted && grpc.ErrorDesc(err) == util.ErrUserSeriesLimitExceeded.Error() {
+			err = util.ErrUserSeriesLimitExceeded
+		}
```

Comment:
I see what you mean; this is yucky.  Would it make sense to:
- propagate this error directly
- remove the "need n successful writes" error and propagate any error directly in that place?
- put all the error-mapping logic in http_server.go?